pkg/service/roomservice.go

```diff
@@ -305,11 +305,6 @@ func (s *RoomService) UpdateRoomMetadata(ctx context.Context, req *livekit.Updat
 		return nil, twirpAuthError(err)
 	}
 
-	room, internal, err := s.roomStore.LoadRoom(ctx, livekit.RoomName(req.Room), false)
-	if err != nil {
-		return nil, err
-	}
-
 	// no one has joined the room, would not have been created on an RTC node.
 	// in this case, we'd want to run create again
 	room, created, err := s.roomAllocator.CreateRoom(ctx, &livekit.CreateRoomRequest{
```

Comment:
When is LoadRoom failing? Seems like we're creating a new room so the behaviour may be different now?

Reply:
LoadRoom will fail if the room doesn't exist. So currently, UpdateRoomMetadata will fail if the room doesn't exist. However, right after the code that returns the error in this case, there is a bunch of code that is meant to handle the case where the room doesn't exist yet by creating it. This seems inconsistent?

Either we make sure that `UpdateRoomMetadata` does create the room as needed, or we remove the whole (currently dead) handling of the case where the room doesn't exist. I'm sot certain what the desired behavior is, but it looks that at least at some point creating the room was desired...

Reply:
I don't think we want to create a room if it doesn't exist when updating the metadata? I may be wrong tho cc @livekit/sfu-devs